modules/ugdk-core/src/system/engine.cc

```diff
@@ -177,7 +177,8 @@ void Run() {
     while (current_state_ == UGDKState::RUNNING) {
         // Start the frame time as first thing
         double delta_t;
-        if (auto time_manager = time::manager()) {
+        if (time::is_active()) {
+            auto time_manager = std::unique_ptr<time::Manager>(new time::Manager);
```

Comment:
You shouldn't create a new `time::Manager`. As the old code did, you have to get from `time::manager()`, as in:

```c++
time::Manager& time_manager = time::manager();
```